### PR TITLE
Initial commit of streaming Kendall's Tau algorithm.

### PR DESCRIPTION
**Pull request by sorensenjs**
# Description

Brief Description of the PR:
Introduces a streaming Kendall's Tau metric implementation.

Fixes # (issue)

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ x ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [ ] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ x ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

File: tensorflow_addons/metrics/kendalls_tau.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+from tensorflow_addons.utils.types import AcceptableDTypes
+
+from typeguard import typechecked
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class KendallsTau(Metric):
+    """Computes Kendall's Tau-b Rank Correlation Coefficient.
+
+    A measure of ordinal similarity between equa length sequences
```
Comment (bhack) [inline tensorflow_addons/metrics/kendalls_tau.py:28]: equal?

File: tensorflow_addons/metrics/kendalls_tau.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        u = (
+            tf.math.reduce_sum(tf.math.square(tf.sparse.to_dense(self.ncol)))
+            - sum_m_squard
+        ) / 2.0
+        # Ties in both.
+        b = tf.math.reduce_sum(tf.multiply(m_dense, (m_dense - 1.0))) / 2.0
+        # Number of discordant pairs.
+        n = tf.cast(self.n, tf.float32)
+        q = (n - 1.0) * n / 2.0 - p - t - u - b
+        print("%s %s %s %s %s" % (p, q, t, u, b))
```
Comment (bhack) [inline tensorflow_addons/metrics/kendalls_tau.py:166]: Is this a debug print?

File: tensorflow_addons/metrics/kendalls_tau.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        Args:
+          y_true: actual rank values.
+          y_pred: predicted rank values.
+          sample_weight (optional): Ignored.
+
+        Returns:
+          Update op.
+        """
+        if y_true.shape and y_true.shape[0]:
+            print("update_state %s %s" % (y_true, y_pred))
```
Comment (bhack) [inline tensorflow_addons/metrics/kendalls_tau.py:93]: Are these debug prints?

File: tensorflow_addons/metrics/kendalls_tau.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        # Ties in y.
+        u = (
+            tf.math.reduce_sum(tf.math.square(tf.sparse.to_dense(self.ncol)))
+            - sum_m_squard
+        ) / 2.0
+        # Ties in both.
+        b = tf.math.reduce_sum(tf.multiply(m_dense, (m_dense - 1.0))) / 2.0
+        # Number of discordant pairs.
+        n = tf.cast(self.n, tf.float32)
+        q = (n - 1.0) * n / 2.0 - p - t - u - b
```
Comment (bhack) [inline tensorflow_addons/metrics/kendalls_tau.py:165]: I don't know if I lost some pass but is in 16 `n+1`?

File: tensorflow_addons/metrics/tests/kendall_tau_test.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    kp = KendallsTau()
+    inputs = tf.keras.layers.Input(shape=(10,))
+    outputs = tf.keras.layers.Dense(1, activation="sigmoid")(inputs)
+    model = tf.keras.models.Model(inputs, outputs)
+    model.compile(optimizer="sgd", loss="binary_crossentropy", metrics=[kp])
+
+    x = np.random.rand(1000, 10).astype(np.float32)
+    y = np.random.rand(1000, 1).astype(np.float32)
+
+    model.fit(x, y, epochs=1, verbose=0, batch_size=32)
```
Comment (bhack) [inline tensorflow_addons/metrics/tests/kendall_tau_test.py:93]: Do we need to check something here after fit?

File: tensorflow_addons/metrics/kendalls_tau.py (hunk 1 above)
Comment (sorensenjs) [reply tensorflow_addons/metrics/kendalls_tau.py:28]: Done

File: tensorflow_addons/metrics/kendalls_tau.py (hunk 2 above)
Comment (sorensenjs) [reply tensorflow_addons/metrics/kendalls_tau.py:166]: Sorry, removed.

File: tensorflow_addons/metrics/kendalls_tau.py (hunk 3 above)
Comment (sorensenjs) [reply tensorflow_addons/metrics/kendalls_tau.py:93]: Removed.

File: tensorflow_addons/metrics/tests/kendall_tau_test.py (hunk 5 above)
Comment (sorensenjs) [reply tensorflow_addons/metrics/tests/kendall_tau_test.py:93]: No, it's not meaningful, this is following the example in cohens_kappa_test, and only tests that fit executes without error.

File: tensorflow_addons/metrics/kendalls_tau.py (hunk 4 above)
Comment (sorensenjs) [reply tensorflow_addons/metrics/kendalls_tau.py:163]: This is the number of ways to choose two from n items is (n-1)*n / 2, that's the total number of pairs. I think this is correct, and it wouldn't produce the right scores if it wasn't.

Comment (bhack) [reply tensorflow_addons/metrics/kendalls_tau.py:163]: It was just cause 16 is:immagine

Comment (sorensenjs) [reply tensorflow_addons/metrics/kendalls_tau.py:163]: One based versus zero based indexing.

**Review by bhack — DISMISSED** [review]

**Comment by bhack** [comment]
Just a final edit, can you add yourself in https://github.com/tensorflow/addons/blob/master/.github/CODEOWNERS for this?

**Review by bhack — DISMISSED** [review]

**Review by bhack — APPROVED** [review]
